docs/tutorials/lattice-fermions.ipynb

```diff
@@ -0,0 +1,762 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #20.                phi = kernel[:, jnp.where(n, size=self.hilbert.n_fermions)[0]]
</pre><p> </p><p>It took me a moment to figure out this line. I don't like it (but I'm not sure how to do it nicely). Maybe it's worth adding a comment saying that jnp where gives you the indices of the orbitals and then you index? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/netket/netket/pull/1581/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Reply:
yes I am using now `nonzero`, hopefully this is clearer? by some arcane reason this is also faster than using where...

Reply:
Much clearer now, thanks.
I suspect the speedup is because nonzero uses some masking tricks, while jnp.where uses a find-like kernel that is probably better suited for very large arrays but very bad for the small ones we use.